src/utils/serial.rs

```diff
@@ -66,6 +66,11 @@ impl SerialCounter {
     pub fn next_serial(&self) -> Serial {
         Serial(self.serial.fetch_add(1, Ordering::AcqRel))
     }
+
+    /// Checks if a serial was generated within the last `max_age` values
+    pub fn is_recent(&self, serial: Serial, max_age: u32) -> bool {
+        self.serial.load(Ordering::Acquire) - max_age < serial.0
+    }
```

Comment:
It's better than nothing and we currently don't have any timestamps attached to serials or a good way to check which event a serial did belong to.

But sure, I can change the api to accept a second lower-bound serial instead.

Reply:
Yeah, I'm pretty sure the `pointer`/`keyboard`/`touch` should have a way to get their serials, at least of the last enter events, since they are stored already internally.

Reply:
I'll open an issue to track this.